src/rgw/rgw_sal_motr.cc

```diff
@@ -4215,7 +4219,9 @@ int MotrMultipartUpload::complete(const DoutPrefixProvider *dpp,
   // Update the dir entry and insert it to the bucket index so
   // the object will be seen when listing the bucket.
   bufferlist update_bl, old_check_bl;
-  target_obj->get_key().get_index_key(&ent.key);  // Change to offical name :)
+  rgw_obj_key key = target_obj->get_key();
+  ent.key.name = key.name;  // Change to offical name :)
+  ent.key.instance = key.instance;
```

Comment:
We are repeating same 3 lines at 3 places, not sure if we can create a function for this.

Reply:
I tried that but `target_obj` is a base class pointer. So creating function in MotrObject class will not work without making it a virtual function which requires changes in the base class.
The other way is to create a utility function that I deferred over this implementation.

Reply:
> The other way is to create a utility function that I deferred over this implementation.

We can skip this as of now, but we will need it for larger functions as we go along.